app/models/manageiq/providers/ibm_cloud/vpc/cloud_manager/provision/cloning.rb

```diff
@@ -0,0 +1,61 @@
+module ManageIQ::Providers::IbmCloud::VPC::CloudManager::Provision::Cloning
+  # Called during provision. Prints message to log file.
+  # @param clone_options [Hash] Options used for clone operation.
+  # @return [Boolean] The result of the log message.
+  def log_clone_options(clone_options)
+    json_clone_options = JSON.dump(clone_options)
+    _log.info("IBM SERVER PROVISIONING OPTIONS: #{json_clone_options}")
+  rescue => e
+    logger(__method__).log_backtrace(e, :context_msg => 'Error while logging clone options for VPC provision.')
+  end
+
+  # Send the final product to the Cloud provider.
+  # @param clone_options [Hash] Payload to send to provider.
+  # @raise [MiqException::MiqProvisionError] An error was returned by the SDK.
+  # @return [String] The ID of the new instance.
+  def start_clone(clone_options)
+    logger(__method__).debug("Json for VPC provision task. #{JSON.dump(clone_options)}")
+    response = source.with_provider_object { |vpc| vpc.request(:create_instance, :instance_prototype => clone_options) }
+    if response[:id].nil?
+      error_msg = _('An error occurred while requesting the IBM VPC instance provision. Cannot retrieve instance id from returned server response.')
+      raise MiqException::MiqProvisionError, error_msg
+    end
+
+    response[:id]
+  rescue IBMCloudSdkCore::ApiException => e
+    raise MiqException::MiqProvisionError, e.to_s
+  rescue => e
+    logger(__method__).log_backtrace(e)
+  end
+
+  # Check the status of the provision.
+  # @param clone_task_ref [String] The UUID for the new provision.
+  # @return [Array(Boolean, String)] 2 elements first is boolean when true signals the provision is complete. Second element is a string for logging the current status.
+  def do_clone_task_check(clone_task_ref)
+    instance = source.with_provider_object { |vpc| vpc.request(:get_instance, :id => clone_task_ref) }
+    live_status = instance[:status]
+    return false, _('IBM VPC instance provision has no status present.') if live_status.nil?
+
+    status = live_status.downcase
+    if status == 'failed'
+      error_msg = _('An error occurred while provisioning the IBM VPC instance. Instance has failed status. Check the instance in cloud.ibm.com for more information.')
+      raise MiqException::MiqProvisionError, error_msg
+    elsif status == 'running'
+      return true, _('The IBM VPC instance has been provisioned and has a running status.')
+    elsif %w[pausing pending restarting resuming starting stopping].include?(status)
+      nil # NoOp: Let the case fall to the end and use the default status.
+    else # If we get here then the status is unknown. We should log the oddity and then let the provision try again.
+      warn_msg = "Unknown IBM VPC instance status received from the cloud API: '#{status}'"
+      method_log.warn(warn_msg)
+    end
+
+    return false, _('The IBM VPC instance is being provisioned.')
+  rescue => e
```

Comment:
This doesn't re-raise the error, so what will happen (particularly with the `raise MiqException::MiqProvisionError` up above which I'd expect to bubble up)

Reply:
2 different uses:
1. A job calls `start_clone` to do the provision.  It completes and returns to the job.
2. The job then uses `do_clone_task_check`  to check the progress of the provision.

The job logic requires different returns.

Reply:
Oh sorry, where you asking about the `log_backtrace`?. The method reraises after logging the exception unless told not to.

Reply:
Not sure I follow.  If these methods don't bubble up their failures, what will happen?

Typically loggers return `true` (or a truthy value) when they log something, so this method will actually return `true` on a failure, which I think will break the flows.  Looks like we might need some failure case specs?

Reply:
I commented [over there](https://github.com/ManageIQ/manageiq-providers-ibm_cloud/pull/198/files#r624117724)  (didn't realize you replied here when I noticed that, and didn't realize it was basically the same conversation thread).

Reply:
I agree a logging method re-raising an exception is highly unexpected (at least within our code base), think we should reevaluate in a follow-up.  Personally I'd take an extra line of code in the exception handler as long as it is obvious what is being done if you don't already know how this particular `log_backtrace` method behaves.

Reply:
Huh? This method is to poll the provision. There are 3 states 
1. false = keep polling
2. true = provision complete
3. Unhandled exception = error but I don't know what to do with it. Fail the provision and log the error.

Reply:
We're not about this method specifically, but rather the overall behavior of `log_backtrace` in the `LoggingMixin`

Reply:
The behaviour is the same but more exact, which is what you want when you are debugging. In many cases during my testing the exception was printed to the screen or otherwise swallowed up because of subsequent error handling. When debugging not having the exact error and where it came from wastes time. So if you are in favour of wasting people's time I can't help you. I can ensure that my code is easy to debug.